OOP/oop_home_task.py

```diff
@@ -0,0 +1,475 @@
+        super().__init__(city, publication_type, publication_date, publication_text)
+
+    def publish_news(self):
+        """
+        Blank method for publishing of news that will be overwritten
+        @return: nothing
+        """
+        pass
+
+
+class Advertisement(Publication):
+    """
+    Class that represents the Advertisement publication type. Inherits Publication class.
+    """
+    def __init__(self, publication_date, expiration_date, publication_text, publication_type=2, city=''):
+        """
+        Method is constructor for Advertisement class
+        @param publication_date: date of publication
+        @param expiration_date: date of Advertisement expiration
+        @param publication_text: text of publication
+        @param publication_type: type of publication
+        @param city: city where something happened
+        """
+        self.advertisement_splitter = 'Private Ad -------------------'
+        self.expiration_date = expiration_date
+        super().__init__(publication_date, publication_type, city, publication_text)
+
+    @staticmethod
+    def calculate_expiration_days(expiration_date):
+        """
+        Method for expiration days calculation. Catches TypeError exception.
+        @param expiration_date: date of Advertisement expiration
+        @return: count of days from today to expiration date
+        """
+        try:
+            expiration_days = (datetime.date.today() - datetime.date.fromisoformat(expiration_date))*-1
+            return expiration_days.days
+        except TypeError:
+            print('Expiration days calculation error.')
+
+    def publish_advertisement(self):
+        """
+        Blank method for publishing of Advertisement that will be overwritten
+        @return: nothing
+        """
+        pass
+
+
+class Article(Publication):
+    """
+    Class that represents the Article publication type. Inherits Publication class.
+    """
+    def __init__(self, publication_text, author, publication_date=None, publication_type=3, city=None):
+        """
+        Method is constructor for Advertisement class
+        @param publication_text: text of publication
+        @param author: author of publication
+        @param publication_date: date of publication
+        @param publication_type: type of publication
+        @param city: city where something happened
+        """
+        self.author = author
+        self.article_splitter = 'Article ----------------------'
+        super().__init__(publication_date, publication_type, city, publication_text)
+
+    @staticmethod
+    def calculate_symbols_count(publication_text):
+        """
+        Method that calculated total of alphanumeric symbols in text entered as Article.
+        @param publication_text: text of publication
+        @return: total symbols count
+        """
+        symbols_count = 0
+        for symbol in publication_text:
+            if symbol.isalnum():
+                symbols_count += 1
```

Comment:
You can try to use 'filter' built-in function for this. It's not obligatory, but JFYI.